server/src/main/java/org/opensearch/index/mapper/BooleanFieldMapper.java

```diff
@@ -257,9 +272,77 @@ public DocValueFormat docValueFormat(@Nullable String format, ZoneId timeZone) {
             return DocValueFormat.BOOLEAN;
         }
 
+        @Override
+        public Query termQuery(Object value, QueryShardContext context) {
+            failIfNotIndexedAndNoDocValues();
+            Query query = new TermQuery(new Term(name(), indexedValueForSearch(value)));
+            if (boost() != 1f) {
+                query = new BoostQuery(query, boost());
+            }
+            if (isSearchable() && hasDocValues()) {
+                return new IndexOrDocValuesQuery(
+                    query,
+                    SortedNumericDocValuesField.newSlowExactQuery(name(), Values.TRUE.bytesEquals(indexedValueForSearch(value)) ? 1 : 0)
+                );
+            }
+            if (hasDocValues()) {
+                return SortedNumericDocValuesField.newSlowExactQuery(name(), Values.TRUE.bytesEquals(indexedValueForSearch(value)) ? 1 : 0);
+            }
+            return query;
+        }
+
+        @Override
+        public Query termsQuery(List<?> values, QueryShardContext context) {
+            failIfNotIndexedAndNoDocValues();
+            boolean seenTrue = false;
+            boolean seenFalse = false;
+            for (Object value : values) {
+                if (Values.TRUE.equals(indexedValueForSearch(value))) seenTrue = true;
+                else seenFalse = true;
+            }
+            if (seenTrue) {
+                if (seenFalse) {
+                    return new MatchAllDocsQuery();
+                }
+                return termQuery(Values.TRUE, context);
+            }
+            if (seenFalse) {
+                return termQuery(Values.FALSE, context);
+            }
+            return new MatchNoDocsQuery("Values did not contain True or False");
```

Comment:
Can be written better as switch case IMO? We can use 2 bit and match these cases with 0, 1, 2, 3?

Reply:
The case here has to switch over a `BytesRef` which seemed to be more complicated so I opted for a simple if